source/result.js

```diff
@@ -101,12 +98,14 @@ Object.assign(Ok.prototype, {
     return this.value;
   },
 
-  map(λ) {
-    return transformResult(() => λ(this.value), Ok);
-  },
-
-  flatMap(λ) {
-    return transformResult(() => λ(this.value), Result.expect);
+  chain(λ) {
+    try {
```

Comment:
DecideWrapper should encapsulate *all* of the logic to transform values. I see why you need that, but it's a bit weird to split the logic in two places (the try catch and decideWrapper)

Reply:
good point